src/Tools/AI Test Toolkit/src/Logs/AITLogEntry.Codeunit.al

```diff
@@ -5,21 +5,163 @@
         AITLogEntry.Run();
     end;
+
+    procedure UpdateRunHistory(Code: Code[100]; LineNo: Integer; AITViewBy: Enum "AIT Run History - View By"; var TempAITRunHistory: Record "AIT Run History" temporary)
+    var
+        AITRunHistory: Record "AIT Run History";
+        SeenTags: List of [Text[20]];
+    begin
+        TempAITRunHistory.DeleteAll();
+        AITRunHistory.SetRange("Test Suite Code", Code);
+
+        if AITViewBy = AITViewBy::Version then
+            if AITRunHistory.FindSet() then
+                repeat
+                    TempAITRunHistory.TransferFields(AITRunHistory);
+                    TempAITRunHistory.Insert();
+                until AITRunHistory.Next() = 0;
+
+        if AITViewBy = AITViewBy::Tag then
+            if AITRunHistory.FindSet() then
+                repeat
+                    if not SeenTags.Contains(AITRunHistory.Tag) then begin
+                        TempAITRunHistory.TransferFields(AITRunHistory);
+                        TempAITRunHistory.Insert();
+                    end;
+                    SeenTags.Add(AITRunHistory.Tag);
+                until AITRunHistory.Next() = 0;
+
+        if (LineNo <> 0) then
+            TempAITRunHistory.SetRange("Line No. Filter", LineNo)
+    end;
+
+    procedure LookupTestMethodLine(TestSuiteCode: Code[100]; var LineNoFilter: Text; var LineNo: Integer)
+    var
+        AITTestMethodLine: Record "AIT Test Method Line";
+        AITTestMethodLines: Page "AIT Test Method Lines Lookup";
+    begin
+        AITTestMethodLine.SetRange("Test Suite Code", TestSuiteCode);
+
+        AITTestMethodLines.SetTableView(AITTestMethodLine);
+        AITTestMethodLines.LookupMode(true);
+
+        if not (AITTestMethodLines.RunModal() = Action::LookupOK) then
+            exit;
+
+        AITTestMethodLines.GetRecord(AITTestMethodLine);
+
+        AITTestMethodLine.CalcFields("Codeunit Name");
+        LineNoFilter := StrSubstNo(LineNoFilterLbl, AITTestMethodLine."Codeunit ID", AITTestMethodLine."Codeunit Name", AITTestMethodLine."Input Dataset");
+        LineNo := AITTestMethodLine."Line No.";
+    end;
+
+    procedure UpdateTestInput(TestInput: Text; TestInputView: Enum "AIT Test Input - View"): Text
+    var
+        TestData: Codeunit "Test Input Json";
+    begin
+        InitTestData(TestInput, TestData);
+
+        case TestInputView of
+            TestInputView::"Full Input":
+                exit(TestInput);
+            TestInputView::Question:
+                exit(GetTestDataElement('question', TestData));
+            TestInputView::Context:
+                exit(GetTestDataElement('context', TestData));
+            TestInputView::"Test Setup":
+                exit(GetTestDataElement('test_setup', TestData));
+            TestInputView::"Ground Truth":
+                exit(GetTestDataElement('ground_truth', TestData));
+            TestInputView::"Expected Data":
+                exit(GetTestDataElement('expected_data', TestData));
```

Comment:
Good candiate for using an interface instead. The seems like something that should be an Enum, then we can add to it as requirements expand.

Reply:
It is already an enum. I think implementing an interface for this scenario seems a little over-engineered. Should I then have 7 implementation codeunits with a single 1-line function? Or what is your suggestion here?

Reply:
Resolving as it's already an enum and implementing an interface and 7 implementations seems over-engineered for this purpose. Please reopen if you disagree.